iree/compiler/Conversion/LinalgToLLVMGPU/KernelConfig.cpp

```diff
@@ -16,18 +16,27 @@ using namespace mlir::iree_compiler;
 
 static constexpr unsigned cudaWarpSize = 32;
 
-static LaunchConfig getOpLaunchConfig(linalg::GenericOp op) {
-  LaunchConfig config;
+static void setConfig(TileSizesListType tileSizes, Operation* op) {
+  IREE::HAL::LoweringConfig config =
+      getConfigAttr(tileSizes, ArrayRef<int64_t>{}, op->getContext());
+  setLoweringConfig(op, config);
+}
+
+static IREE::HAL::TranslateExecutableInfo setRootConfig(linalg::GenericOp op) {
+  IREE::HAL::TranslateExecutableInfo info;
+  info.passPipeline = IREE::HAL::DispatchLoweringPassPipeline::LLVMGPUVectorize;
+  TileSizesListType tileSizes;
   size_t numLoops = getNumOuterParallelLoops(op);
   if (numLoops == 0) {
     // Pure reduction, we serialize the operation on a single thread.
     // TODO: Use atomic to allow distributing reduction loops.
-    config.setWorkgroupSize({1, 1, 1});
-    config.setTileSizes(op, {}, 0);
-    return config;
+    info.workgroupSize.append({1, 1, 1});
+    tileSizes.push_back({});
+    setConfig(tileSizes, op);
+    return info;
   }
 
-  config.setWorkgroupSize({cudaWarpSize, 1, 1});
+  info.workgroupSize.append({cudaWarpSize, 1, 1});
```

Comment:
I would just use `assign` here. `append` might have unintended consequence.

Reply:
This code is now gone after rebase.